loader/unknown_ext_chain_gas_aarch.S

```diff
@@ -55,6 +55,7 @@
 
 .macro PhysDevExtTramp num
 .global vkPhysDevExtTramp\num
+.type vkPhysDevExtTramp\num, @function
```

Comment:
This change is causing the MacOS CI to fail - seems that github actions CI is running on x86_64 so isn't affected. 

I can send you the full log, but the main error is this - related to the `.type vkPhysDevExtTramp\num, @function`

```
<instantiation>:2:1: error: unknown directive
.type vkPhysDevExtTramp0, @function
^
/Users/lunarg/.jenkins/w3/Release64-3/Vulkan-Loader/loader/unknown_ext_chain_gas_aarch.S:208:5: note: while in macro instantiation
    PhysDevExtTramp 0
    ^
```

Reply:
```
loading initial cache file /Users/lunarg/.jenkins/w3/Release64-3/Vulkan-Loader/external/helper.cmake
-- The C compiler identification is AppleClang 15.0.0.15000040
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /usr/bin/git (found version "2.39.3 (Apple Git-145)") 
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Found PkgConfig: /opt/homebrew/bin/pkg-config (found version "0.29.2") 
-- The CXX compiler identification is AppleClang 15.0.0.15000040
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /opt/homebrew/Frameworks/Python.framework/Versions/3.10/bin/python3.10 (found version "3.10.8") found components: Interpreter 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/lunarg/.jenkins/w3/Release64-3/Vulkan-Loader/build
```

The CMake configure indicates what version of the compiler is being used, as well as the assembler used.
-- The C compiler identification is AppleClang 15.0.0.15000040
-- The ASM compiler identification is Clang with GNU-like command-line

So apparently the `.type` directive isn't supported with apple clang? Or its malformed? I'll look into it and see if there is a 'quick fix'.

Reply:
Oh, I didn't realize it wouldn't work on Mac. We could probably move it inside `#if defined(__ELF__)` below; it's not critical to this patch.

Reply:
I'm happy to implement the move, just not wanting to step on your toes if you are going to move the .type into the #if defined(__ELF__) blocks.